### PR TITLE
create delete button and modal for desired state versions with test c…

### PR DESCRIPTION
**Pull request by matborowczyk**
…overage

# Description

* Short description here *

closes *Add ticket reference here*

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
~~- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/Slices/DesiredState/UI/Components/Actions.tsx [hunk 1]
```diff
@@ -20,6 +21,7 @@ export const Actions: React.FC<Props> = ({ version, isPromoteDisabled }) => {
       position="right"
       onSelect={() => setIsOpen(false)}
       dropdownItems={[
+        <DeleteAction key="delete" version={version} />,
```
Comment (LauraStordeur) [inline src/Slices/DesiredState/UI/Components/Actions.tsx:24]: why is it cast to a Number in line 30, and not in the other places?

Comment (matborowczyk) [reply src/Slices/DesiredState/UI/Components/Actions.tsx:24]: unfortunately, I don't know, as these components were there from the start, maybe that made simpler to compare version to each other, but it's only a guess.
I'll try to dig and check what was the reasoning behind that :D

File: src/Slices/DesiredState/UI/Page.test.tsx [hunk 2]
```diff
@@ -462,3 +464,108 @@ test("DesiredStatesView shows CompileWidget", async () => {
...
+    apiHelper.resolve(204);
+
+    await act(async () => {
+      await apiHelper.resolve(Either.right(DesiredStateVersions.response));
+    });
+    const rows = await screen.findAllByRole("row", {
+      name: "DesiredStates Table Row",
+    });
+
+    //close by "no" button scenario
```
Comment (LauraStordeur) [inline src/Slices/DesiredState/UI/Page.test.tsx:510]: you mean clicking outside the container of the modal?

Comment (LauraStordeur) [reply src/Slices/DesiredState/UI/Page.test.tsx:510]: haaa okay, the button with "no" written in it xD

Comment (matborowczyk) [reply src/Slices/DesiredState/UI/Page.test.tsx:510]: exactly :D

**Review by LauraStordeur — APPROVED** [review]

**Pull request closed without merging**